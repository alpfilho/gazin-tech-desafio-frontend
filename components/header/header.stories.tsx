import { ComponentMeta, ComponentStory } from '@storybook/react';

import { Header } from './header';

export default {
	title: 'Components/Header',
	component: Header,
	argTypes: {
		themeSwitch: {
			type: 'function'
		}
	},
	parameters: {
		layout: 'fullscreen'
	}
} as ComponentMeta<typeof Header>;

const themeSwitch = () => {
	console.log('theme switch');
};

export const Default: ComponentStory<typeof Header> = (args) => <Header {...args} />;

Default.args = {
	activeTheme: 'dark',
	themeSwitch
};
