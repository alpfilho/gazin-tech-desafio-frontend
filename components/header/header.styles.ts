import styled from 'styled-components';
import { Container, Content } from 'components/contentContainer';

export const headerHeight = '6em';

export const HeaderContainer = styled.header`
	width: 100%;
	height: ${headerHeight};
	background-color: ${(props) => props.theme.background};
	transition: background-color 250ms linear;
`;

export const HeaderContent = styled.div`
	position: fixed;
	background-color: ${(props) => `${props.theme.background}f2`};
	transition: background-color 250ms linear;

	z-index: 10;
	width: 100%;
	height: ${headerHeight};

	box-shadow: 0 1em 10em rgba(0, 0, 0, 0.33);

	${Container} {
		height: 100%;
		width: 100%;
		/* background-color: red; */
		display: flex;
		flex-direction: row;
		justify-content: center;
		align-items: center;

		${Content} {
			display: flex;
			flex-direction: row;
			justify-content: space-between;
			align-items: center;

			@media screen and (max-width: 768px) {
				justify-content: center;
			}
		}
	}
`;

export const LogoContainer = styled.a`
	display: block;
	height: auto;
	width: 100%;
	max-width: 20em;

	@media screen and (max-width: 920px) {
		max-width: 40%;
	}

	@media screen and (max-width: 768px) {
		display: none;
	}
`;

export const RightContainer = styled.div`
	display: flex;
	flex-direction: row;
	justify-content: center;
	align-items: center;
`;
