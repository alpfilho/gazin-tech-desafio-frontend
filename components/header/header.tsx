import React from 'react';
import { useTheme } from 'styled-components';

import { Logo } from '@components/logo';

import { HeaderContainer, LogoContainer } from './header.styles';
import Link from 'next/link';

export const Header: React.FC = () => {
	const theme = useTheme();

	return (
		<HeaderContainer>
			<Link href="/" passHref>
				<LogoContainer>
					<Logo color={theme.colors.white} />
				</LogoContainer>
			</Link>
		</HeaderContainer>
	);
};
