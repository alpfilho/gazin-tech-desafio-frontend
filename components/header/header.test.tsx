import { composeStories } from '@storybook/testing-react';
import { render, screen } from '@testing-library/react';

jest.mock('next/router', () => require('next-router-mock'));

import * as stories from './header.stories';

const { Default } = composeStories(stories);

describe('Header', () => {
	it('render a header', () => {
		render(<Default />);
		const header = screen.getByTestId('header');
		expect(header).toBeInTheDocument();
	});

	it('header render unchanged', () => {
		const { container } = render(<Default />);

		expect(container).toMatchSnapshot();
	});
});
