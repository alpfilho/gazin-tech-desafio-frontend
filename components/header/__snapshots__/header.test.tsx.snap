// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header header render unchanged 1`] = `
<div>
  <header
    class="HeaderContainer-sc-17fiyp8-0 eDZhvJ"
    data-testid="header"
  >
    <div
      class="HeaderContent-sc-17fiyp8-1 khrzF"
    >
      <div
        class="Container-sc-yq2cgn-0 leaVgY"
      >
        <div
          class="Content-sc-yq2cgn-1 etbrxE"
        >
          <a
            class="LogoContainer-sc-17fiyp8-2 hZxsph"
            href="/home"
          >
            <svg
              class="LogoContainer-sc-1qw03iv-0 leVVGK"
              data-testid="gazin-logo"
              fill="none"
              viewBox="0 0 909 111"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M71.64 85.9h58.3l31.6-55.97h-44.81l-8.12 14.2 13.53-.11-15.1 26.28s-10.83-.22-11.64-.15c-.81.08-7.79-.61-9.27-1-10.45-2.76-13.21-8.17-14.15-12.65-1.7-8.07 2.65-14.79 2.65-14.79C80.58 31.47 91.13 24.75 93.97 23c9.8-6.07 20.1-7.45 24.44-8.04l1.19-.16c3.3-.5 9.55-.56 9.55-.56h231.52L368.64 0H137.29c-21.72 1.2-32.42 2.97-35.02 3.4l-.23.04C81 6.87 65.67 14.96 63.06 16.43 38.69 30.1 38.93 47.79 38.98 50.98v.11c.26 21.02 19.64 30.7 28.46 34.02 2.6.98 4.2.8 4.2.8ZM160.1 111c-6.1-3.88-10.82-7.54-13.29-14.54H8.2L0 111h160.1Z"
                fill="#282424"
              />
              <path
                d="m257.5 36.38 9.12-16.14h-7.49s-16.38-.13-23.28.36c-24.6 1.75-50.89 8.77-69.98 25.78l-.09.07-.18.17-.09.07c-1.92 1.66-8.08 6.98-13.3 17.47-1.03 2.08-4.37 10.7.08 21.5l.01.03.04.08c.64 1.58 4.48 11.06 19.64 21.95l13.22-14.4s-5.75-4.29-7.33-10.05a16.2 16.2 0 0 1 1.44-10.7l.24-.48c.45-.9.9-1.8 1.48-2.6l28.07-.01-9.25 16.42h29.8l24.5-43.53h-29.79l-5.93 10.53-27.07-.02s5.82-10.96 28.16-15.16c1.97-.37 10-1.34 16.99-1.34h21Zm9.3-.31 8.5-15.83h81.96l-63.51 49.28h35.77l-9.23 16.39H238.3l64.06-49.84H266.8Zm71.88 50.22 36.6-65.68h36.46L374.99 86.3h-36.3Zm39.63-72.09h36.28l8.1-14.2h-36.33l-8.05 14.18ZM434.56 0l-8.02 14.23H530.9L535.8 4h25l-4.9 10.23h23L583.8 4h25l-4.9 10.23h23L631.8 4h25l-4.9 10.23h23L679.8 4h25l-4.9 10.23h23L727.8 4h25l-4.9 10.23h23L775.8 4h25l-4.9 10.23h23L823.8 4h25l-4.9 10.23h23L871.8 4h25l-4.9 10.23h9.1l8-14.22H434.56Z"
                fill="#282424"
              />
              <path
                d="m386.2 85.9 36.95-65.66h36.05l7.07 21.92 12.35-21.92h34L469.37 97h12.53l-4.52 9.64h25L506.9 97h23l-4.52 9.64h25L554.9 97h23l-4.52 9.64h25L602.9 97h23l-4.52 9.64h25L650.9 97h23l-4.52 9.64h25L698.9 97h23l-4.52 9.64h25L746.9 97h23l-4.52 9.64h25L794.9 97h23l-4.52 9.64h25L842.9 97h17.59l-8 14H185.73l8.23-14.54H435.7l16.06-28.52-9.27-25.84-24.66 43.8H386.2Z"
                fill="#282424"
              />
              <path
                d="M827.77 88c-11.83 0-20.83-1.57-27-4.72-6.08-3.15-8.23-7.75-6.45-13.8.18-.6.53-1.51 1.07-2.72h32.21c-.82 1.87-.82 3.26 0 4.17.92.91 2.7 1.36 5.33 1.36 1.64 0 3.09-.27 4.34-.81 1.25-.55 2.03-1.33 2.33-2.36.43-1.46-.3-2.67-2.2-3.63-1.81-.97-4.89-2.12-9.23-3.45a102.24 102.24 0 0 1-11.6-4.18 18.43 18.43 0 0 1-7-5.45c-1.78-2.3-2.16-5.17-1.14-8.62 1.33-4.54 4.16-8.44 8.49-11.71 4.44-3.33 9.95-5.84 16.55-7.54A86.8 86.8 0 0 1 855.1 22c7 0 13.02.73 18.05 2.18 5.13 1.45 8.83 3.57 11.1 6.35 2.38 2.73 3.02 5.97 1.92 9.72-.14.48-.48 1.33-1.02 2.54h-32.62c.4-.73.63-1.21.7-1.45.34-1.15.1-2.03-.73-2.64-.8-.66-2.07-1-3.8-1-1.54 0-2.9.3-4.09.91-1.18.6-1.9 1.4-2.2 2.36-.38 1.34.32 2.52 2.1 3.54a70.9 70.9 0 0 0 8.9 3.64 102.4 102.4 0 0 1 11.47 4.63 18.43 18.43 0 0 1 7.16 5.8c1.76 2.37 2.14 5.27 1.12 8.72-1.15 3.94-3.79 7.48-7.9 10.62-4.02 3.15-9.25 5.63-15.7 7.45-6.43 1.75-13.7 2.63-21.79 2.63ZM582.23 39.07l8.93-15.98h-68.24l-35.96 64.27h30.3l13.54-24.23h27.3l8.55-15.17h-27.3l4.94-8.9h37.94Zm8.07 48.29 35.97-64.27h-30.3L560 87.36h30.3Zm47.1-15.25h29.21l-8.58 15.25h-59.5l35.96-64.27h30.3L637.4 72.1Zm140.85 15.25 35.97-64.27h-36.99l-42.36 38.67.32-38.67h-37.12l-35.96 64.27h30.3l17.94-32.22-3.07 32.22h25.94l32.73-31.95-17.86 31.95h30.16Z"
                fill="#282424"
              />
            </svg>
          </a>
          <div
            class="RightContainer-sc-17fiyp8-3 glVRxP"
          >
            <form
              autocomplete="off"
              class="SearchForm-sc-13brd7v-0 coCkLY"
              novalidate=""
              spellcheck="false"
            >
              <div
                class="InputContainer-sc-13brd7v-1 dyKORw"
              >
                <input
                  class="InputElement-sc-13brd7v-2 kQnvfZ"
                  name="search"
                  placeholder="Busque filmes"
                  type="text"
                />
              </div>
              <button
                class="SearchButton-sc-13brd7v-3 hvQdir"
                type="submit"
              >
                <div
                  class="IconContainer-sc-13brd7v-4 laineT"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-magnifying-glass "
                    color="#282424"
                    data-icon="magnifying-glass"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M500.3 443.7l-119.7-119.7c27.22-40.41 40.65-90.9 33.46-144.7C401.8 87.79 326.8 13.32 235.2 1.723C99.01-15.51-15.51 99.01 1.724 235.2c11.6 91.64 86.08 166.7 177.6 178.9c53.8 7.189 104.3-6.236 144.7-33.46l119.7 119.7c15.62 15.62 40.95 15.62 56.57 0C515.9 484.7 515.9 459.3 500.3 443.7zM79.1 208c0-70.58 57.42-128 128-128s128 57.42 128 128c0 70.58-57.42 128-128 128S79.1 278.6 79.1 208z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </button>
            </form>
            <div
              class="SwitcherContainer-sc-9phzop-0 gsZeyz"
            >
              <div
                class="ThemeNameContainer-sc-1tv9hgu-0 geAmfw"
                style="opacity: 1;"
              >
                Dark
              </div>
              <button
                class="ThemeSwitcherContainer-sc-9phzop-1 iqIoJg"
                title="Alterar para tema claro"
              >
                <div
                  class="SwitcherContent-sc-9phzop-2 ekiWwd"
                  style="transform: translateX(0%) translateZ(0);"
                >
                  <div
                    class="SwitcherIconContainer-sc-9phzop-3 fQfslh"
                    style="opacity: 1; transform: translateX(-50%) translateY(-50%) translateZ(0);"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-moon "
                      color="#f3f1f1"
                      data-icon="moon"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M32 256c0-123.8 100.3-224 223.8-224c11.36 0 29.7 1.668 40.9 3.746c9.616 1.777 11.75 14.63 3.279 19.44C245 86.5 211.2 144.6 211.2 207.8c0 109.7 99.71 193 208.3 172.3c9.561-1.805 16.28 9.324 10.11 16.95C387.9 448.6 324.8 480 255.8 480C132.1 480 32 379.6 32 256z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                  <div
                    class="SwitcherIconContainer-sc-9phzop-3 fQfslh"
                    style="opacity: 0; transform: translateX(-100%) translateY(-50%) translateZ(0);"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-sun "
                      color="#515151"
                      data-icon="sun"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M256 159.1c-53.02 0-95.1 42.98-95.1 95.1S202.1 351.1 256 351.1s95.1-42.98 95.1-95.1S309 159.1 256 159.1zM509.3 347L446.1 255.1l63.15-91.01c6.332-9.125 1.104-21.74-9.826-23.72l-109-19.7l-19.7-109c-1.975-10.93-14.59-16.16-23.72-9.824L256 65.89L164.1 2.736c-9.125-6.332-21.74-1.107-23.72 9.824L121.6 121.6L12.56 141.3C1.633 143.2-3.596 155.9 2.736 164.1L65.89 256l-63.15 91.01c-6.332 9.125-1.105 21.74 9.824 23.72l109 19.7l19.7 109c1.975 10.93 14.59 16.16 23.72 9.824L256 446.1l91.01 63.15c9.127 6.334 21.75 1.107 23.72-9.822l19.7-109l109-19.7C510.4 368.8 515.6 356.1 509.3 347zM256 383.1c-70.69 0-127.1-57.31-127.1-127.1c0-70.69 57.31-127.1 127.1-127.1s127.1 57.3 127.1 127.1C383.1 326.7 326.7 383.1 256 383.1z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </div>
              </button>
              <div
                class="ThemeNameContainer-sc-1tv9hgu-0 geAmfw"
                style="opacity: 0.2;"
              >
                Light
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;
